src/open_producten/utils/oauth.py

```diff
@@ -0,0 +1,37 @@
+from typing import List, Union
+
+from drf_spectacular.extensions import OpenApiAuthenticationExtension, _SchemaType
+from drf_spectacular.openapi import AutoSchema
+from drf_spectacular.plumbing import build_bearer_security_scheme_object
+from mozilla_django_oidc_db.backends import (
+    OIDCAuthenticationBackend as _OIDCAuthenticationBackendDB,
+)
+from mozilla_django_oidc_db.models import OpenIDConnectConfig
+from mozilla_django_oidc_db.typing import JSONObject
+
+
+class OIDCAuthenticationBackend(_OIDCAuthenticationBackendDB):
+    """
+    `mozilla_django_oidc_db.backends.OIDCAuthenticationBackend` only load in the config_class within the authenticate method.
+    The drf integration of the base package does not use this method and instead just calls the userinfo endpoint of the oidc provider.
+
+    This class sets the config_class so that it is accessible in the get_userinfo method.
+    """
+
+    def get_userinfo(
+        self, access_token: str, id_token: str, payload: JSONObject
+    ) -> JSONObject:
+        self.config_class = OpenIDConnectConfig
```

Comment:
Do note that you now use the same singletonmodel that's used for the admin authentication with OIDC, not sure if that could cause problems. In the future mozilla-django-oidc-db will be refactored to use regular models instead, but I'm not sure when that is expected to be done